### PR TITLE
Add GeoJson download task

### PR DESCRIPTION
**Pull request by ovruni**
This fixes the issue #21

### CODE REVIEW

**Comment by don-vip** [comment]
Travis isn't happy. You have to increase the JOSM version in gradle properties file. The missing class has been added in 13927 but this plugin currently requires only 13878. 13929 should work (oldest snapshot available containing this class)

**Comment by ovruni** [comment]
@don-vip Everything seems fine now.

**Comment by don-vip** [comment]
Thanks a lot!

File: src/main/java/org/openstreetmap/josm/plugins/geojson/GeoJsonDownloadTask.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            }
+            if (settings.getUploadPolicy() != null) {
+                dataSet.setUploadPolicy(settings.getUploadPolicy());
+            }
+            if (dataSet.isLocked() && !settings.isLocked()) {
+                dataSet.unlock();
+            } else if (!dataSet.isLocked() && settings.isLocked()) {
+                dataSet.lock();
+            }
+            return new GeoJsonLayer(tr("Data Layer from GeoJSON: ") + layerName, dataSet, null);
```
[the remark this answers is not in the document]
Comment (floscher) [reply src/main/java/org/openstreetmap/josm/plugins/geojson/GeoJsonDownloadTask.java]: fixed in 51b8ab68db2d7a06381e5d543b202f0260f6f59e

File: src/main/java/org/openstreetmap/josm/plugins/geojson/GeoJsonServerReader.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    private final String url;
+
+    public GeoJsonServerReader(String url) {
+        this.url = url;
+    }
+
+    @Override
+    public DataSet parseOsm(ProgressMonitor progressMonitor) throws OsmTransferException {
+        try {
+            progressMonitor.beginTask(tr("Contacting Server...", 10));
```
[the remark this answers is not in the document]
Comment (floscher) [reply src/main/java/org/openstreetmap/josm/plugins/geojson/GeoJsonServerReader.java]: fixed in 51b8ab68db2d7a06381e5d543b202f0260f6f59e

File: src/main/java/org/openstreetmap/josm/plugins/geojson/GeoJsonDownloadTask.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+        private final String url;
+
+        InternalDownloadTask(DownloadParams settings, String url, ProgressMonitor progressMonitor) {
+            super(settings, new GeoJsonServerReader(url), progressMonitor);
+            this.url = url;
+        }
+
+        @Override
+        protected GeoJsonLayer createNewLayer(String layerName) {
```
[the remark this answers is not in the document]
Comment (floscher) [reply src/main/java/org/openstreetmap/josm/plugins/geojson/GeoJsonDownloadTask.java]: fixed in b31f896097fa088a10c493204eaf8c1c15ac00b0

**Pull request merged**